### ISSUE TITLE
Api.count does not apply the policy of the action

### ISSUE THREAD

**Issue by pierrelegall** [ash-project/ash#731]
**Bug description**
`Api.count` and `Api.count!` does not apply the policy of the action.

**To Reproduce**
Having two read actions with different policies, like:

```elixir
  # in MyResource

  policies do
    policy action_type(:read) do
      authorize_if always()
    end

    policy action(:read_updatable) do
      authorize_if relates_to_actor_via([:creator])
    end
  end
```

The problem happens here:

```elixir
MyResource
|> Ash.Query.new()
|> Api.count!(actor: actor, action: :read_updatable)
# => does not return the expected count

# and, does not work even if we specify the action at the query creation:

MyResource
|> Ash.Query.for_read(:read_updatable)
|> Api.count!(actor: actor)
# => does not return the expected count
```

However:

```elixir
MyResource
|> Ash.Query.new()
|> Api.read!(actor: actor, action: :read_updatable)
|> Enum.count()
# => returns the expected count

# works too this way:

MyResource
|> Ash.Query.for_read(:read_updatable)
|> Api.read!(actor: actor)
|> Enum.count()
# => returns the expected count

# or with the code interface:

MyResource.read_updatable!(actor: actor)
|> Enum.count()
# => returns the expected count
```

**Expected behavior**
`Api.count` and `Api.count!` should honor the policy of the action.

Note: it seems to be the same for `Api.exists?`, maybe more?

**Runtime**
 - Elixir version: 1.15.6
 - Erlang version: 26.1.1
 - OS: Archlinux, Linux 6.5.5-arch1-1
 - Ash version: 2.15.10

**Comment by zachdaniel** [comment]
Meant to mark https://github.com/ash-project/ash_postgres/issues/171 as fixed by the linked commit, not this ticket, sorry.

**Comment by zachdaniel** [comment]
Can you try it using `Ash.Query.for_read` also? With specifying explicit `authorize?: true`?

**Comment by zachdaniel** [comment]
Also, try this:
```elixir
%{count: count} = Api.aggregate!(Post, {:count, :count}, authorize?: true)
```

And let me know if that properly applies authorization. I think this is an issue in properly applying the options somewhere.

**Comment by zachdaniel** [comment]
I've got this reproduced now :) Will have a fix shortly.

**Comment by zachdaniel** [comment]
Fixed in v2.5.11

**Comment by pierrelegall** [comment]
> ```elixir
> MyResource
> |> Ash.Query.new()
> |> Api.count!(actor: actor, action: :read_updatable)
> # => does not return the expected count
> 
> # and, does not work even if we specify the action at the query creation:
> 
> MyResource
> |> Ash.Query.for_read(:read_updatable)
> |> Api.count!(actor: actor)
> # => does not return the expected count
> ```

> ```elixir
> %{count: count} = Api.aggregate!(Post, {:count, :count}, authorize?: true)
> ```

This use cases still return an unexpected result with `ash` v2.15.12 and `ash_postgres` v1.3.54 :open_mouth:

In the second one, I do not understand the use of the `authorize?: true` options without specifying an `actor` and an `action` :thinking:

And:

```elixir
%{count: count} = Djo.Api.aggregate!(Event, {:count, :count}, actor: actor, action: :read_manageable)
```

Returns the same unexpected `count`.

**Comment by zachdaniel** [comment]
`nil` is a valid actor, so just specifying `authorize?: true` is like saying "run authorization as a non-logged in user".

Additionally, I you don't specify an action, it will use the primary read action on the resource in question.

Could you clarify for me if you're seeing exactly the same behavior as before for all of the scenarios listed? Or only for `Api.aggregate`?

**Comment by pierrelegall** [comment]
I reproduce the same behavior as explained in the initial issue.

With:

```
$ mix deps | grep ash
* ash 2.15.12 (Hex package) (mix)
  locked at 2.15.12 (ash) 9c240f7f
* ash_authentication 3.11.15 (Hex package) (mix)
  locked at 3.11.15 (ash_authentication) 931926d8
* ash_authentication_phoenix 1.8.3 (Hex package) (mix)
  locked at 1.8.3 (ash_authentication_phoenix) bbb891ed
* ash_phoenix 1.2.19 (Hex package) (mix)
  locked at 1.2.19 (ash_phoenix) a36bc916
* ash_postgres 1.3.54 (Hex package) (mix)
  locked at 1.3.54 (ash_postgres) 82065e46
```

**Comment by zachdaniel** [comment]
Thank you for the detail and for bearing with me :)

This is resolved in v2.15.17

**Comment by pierrelegall** [comment]
A pleasure, thank you for your work and reactivity on Ash :ok_hand:
